src/common/types/value.cpp

```diff
@@ -644,18 +644,25 @@ Value Value::TIMESTAMP(int32_t year, int32_t month, int32_t day, int32_t hour, i
 	return val;
 }
 
-Value Value::STRUCT(child_list_t<Value> values) {
+Value Value::STRUCT(const LogicalType &type, vector<Value> struct_values) {
 	Value result;
+	auto child_types = StructType::GetChildTypes(type);
```

Comment:
A little late to review this, my bad, but still might be worthwhile to mention
Shouldn't we assert that `struct_values.size()` == `child_types.size()`